ambry-store/src/main/java/com.github.ambry.store/Journal.java

```diff
@@ -154,6 +167,23 @@ void addEntry(Offset offset, StoreKey key) {
     return journalEntries;
   }
 
+  /**
+   * @return all journal entries at the time when this method is invoked. (Return empty list if journal is empty)
+   */
+  List<JournalEntry> getAllEntries() {
+    List<JournalEntry> result = new ArrayList<>();
+    // get current last Offset
+    Offset lastOffset = getLastOffset();
+    if (lastOffset != null) {
+      // get portion view of journal whose key is less than or equal to lastOffset
+      NavigableMap<Offset, StoreKey> journalView = journal.headMap(lastOffset, true);
```

Comment:
might be able to just iterate through `journal.entrySet()` instead of finding the last offset on line 176. Not sure if the extra entries returned by doing so would cause any issues

Reply:
I don't see any issues but I am slightly leaning towards current implementation. The `lastOffset` denotes a clear cutoff and headMap only contains keys that below the cutoff. The `entrySet()` keeps changing/growing during the iteration, I feel the last entry is not so deterministic.